system/jlib/jfile.cpp

```diff
@@ -2410,6 +2445,57 @@ IFileAsyncResult *CFileAsyncIO::writeAsync(offset_t pos, size32_t len, const voi
 
 #endif
 
+//---------------------------------------------------------------------------
+
+void FileIOStats::merge(IFileIO * io)
+{
+    if (!io)
+        return;
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        values[stat] += io->getStatistic((FileIOStat)stat);
+}
+
+void FileIOStats::merge(const FileIOStats & other)
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        values[stat] += other.values[stat];
+}
+
+void FileIOStats::merge(MemoryBuffer & in)
+{
+    unsigned __int64 value;
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+    {
+        in.read(value);
+        values[stat] += value;
+    }
+}
+
+void FileIOStats::deserialize(MemoryBuffer & in)
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        in.read(values[stat]);
+}
+
+#if 0
+void FileIOStats::serialize(MemoryBuffer & out) const
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+        out.append(values[stat]);
+}
+#endif
+
+void FileIOStats::serializeNormalized(MemoryBuffer & out) const
+{
+    for (unsigned stat = 0; stat < FIOSmax; stat++)
+    {
+        unsigned __int64 curValue = values[stat];
+        if (stat == FIOSiocycles)
+            curValue = cycle_to_nanosec(curValue);
```

Comment:
The converting it the problem.

- It needs to be converted on the machine that gathered the values - since machines (especially the master) may be running on different chips with different scaling factors.
- What I don't like is that on the master the value has one meaning and on the slave another.
- Note at the moment the fileio is not using the stats enumeration in jstats it is a separate one.

The following is possibly the cleanest solution:
* Switch to using CRuntimeStatisticCollection instead of FileIOStats
* Add an internal Cycles version of each timing stat
* Use CRuntimeStatisticCollection with a mapping class that saves cycle based values
* Have another instance  with time based values
* Have a function that maps between them.

The serialization would then be

````
CRuntimeStatisticCollection localStats(LocalIOMappings);
localStats.merge(timing1);
localStats.merge(timing2);

CRuntimeStatisticCollection remoteStats(RemoteIOMappings);
remoteStats.translate(localStats);
remoteStats.serialize();

````

It would probably be relatively efficient as well.

Disadvantages/Questions
- Slightly more complicated
- Should this also be used for the parameter to the IO class.  If so, should the io class gather reads and writes separately?  My only concern is the extra space, and that is unlikely to be a big issue.

@jakesmith what do you think?  I'm tempted to have a go at changing it.